app/main.py

```diff
@@ -1,8 +1,23 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    people_list = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        if "wife" in person.keys():
+            if person["wife"]:
+                if not hasattr(Person.people[person["name"]], ""):
```

Comment:
There is an issue here with the `hasattr` check. The attribute name should not be an empty string. It should be 'wife' to correctly check if the 'wife' attribute exists on the `Person` instance.